### DISCUSSION TITLE
Set variable from hook

### DISCUSSION THREAD

**Discussion by CaptainQuirk** [AnWeber/vscode-httpyac#212] [General]
Hi ! 

I would like to set a simple `jwt` variable as value of a `Autorization: Bearer` header. My api does not use oauth2 and simply returns a token on a `POST /auth/token` endpoint.

Is it possible to perform the call in a hook and set the value of the jwt variable there ?

Thanks in advance !

**Comment by AnWeber** [comment]
It should go easily. Just use requestHook and set your JWT Token as Variable. See Interfaces [httpClientProvider](https://github.com/AnWeber/httpyac/blob/main/src/models/httpClientProvider.ts#L25) and [HttpHooksApi](https://github.com/AnWeber/httpyac/blob/main/src/models/httpHooksApi.ts#L21)

> .httpyac.js
```javascript
module.exports = {
   configureHooks: function (api) {
      api.hooks.onRequest.addHook('jwt', async function (request, context) {
        const response = await api.httpClientProvider.exchange({method: "POST", url: "/auth/token"}, {});
        request.headers = Object.assign({
          'Authorization':  `Bearer ${response.parsedBody.access_token}',
         }, request.headers);
        context.variables.jwt = response.parsedBody.access_token;
    });
  }
}
```

It may be easier to resolve it with `@ref` and `@import`, but I also use the Hooks approach for the Authorization Header.

**Reply by CaptainQuirk** [reply]
Hi !

Thanks for your response ! It helped a lot !.

I think I got it wrong by trying to add the header line in each request and use a `{{jwt}}` variable as **Bearer** value. Your way makes much more sense and I don't need to set the variable itself.

It's working now !

**Reply by AnWeber** [reply]
Cool. An additional hint. I use the environment variable [`HTTPYAC_PLUGIN`](https://github.com/AnWeber/httpyac/blob/main/src/store/httpFileStore.ts#L149) to automatically load this script in all projects. Maybe I missed documenting this.

```sh
export HTTPYAC_PLUGIN="~/.httpyac.js"
```
